### PR TITLE
Fixes "space panel kebab menu is rendered out of view on sub spaces" 

### PR DESCRIPTION
**Pull request by yaya-usman**
Fixes:  [vector-im/element-web#21141](https://github.com/vector-im/element-web/issues/21141)

Issue reported by @kerryarchibald  

Signed-off-by: Yaya Usman [yaya-usman@users.noreply.github.com](mailto:yaya-usman@users.noreply.github.com) 

**Before:**  (video credit @kerryarchibald )
before

**After:** 

afteer

Type: defect

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fixes "space panel kebab menu is rendered out of view on sub spaces"  ([\#8350](https://github.com/matrix-org/matrix-react-sdk/pull/8350)). Contributed by @yaya-usman.

Preview: https://pr8350--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by yaya-usman** [comment]
> @yaya-usman is this still a draft? If it's ready for review, please click "Ready for review".

I forgot to mark it as ready, Thanks.

**Comment by yaya-usman** [comment]
I had to cut the text early enough with a seperate of concern since the menu button is represented with ellipses as well, it'll make it look ugly and confusing if it's close to the button

**Comment by robintown** [comment]
That change would require design review, so I'd encourage that this PR only try to restore the expected behavior of the current designs, even if that puts the two ellipses close to each other.

**Comment by yaya-usman** [comment]
> That change would require design review, so I'd encourage that this PR only try to restore the expected behavior of the current designs, even if that puts the two ellipses close to each other.

Alright no problem.

**Comment by yaya-usman** [comment]
@robintown How about now?

capture11

**Comment by robintown** [comment]
On my end, this seems to have squashed all space names:

Screenshot 2022-04-20 at 15-22-15 Element Unencrypted test room

**Comment by yaya-usman** [comment]
> On my end, this seems to have squashed all space names:
> 
> Screenshot 2022-04-20 at 15-22-15 Element Unencrypted test room

i guess because you collapsed the space and the name is long, so it automatically puts ellipses. that's the default behavior not related to my own change.

**Comment by yaya-usman** [comment]
Here is mine, home space is having a shorter name, it didn't squash the name. but for longer text in this case it does when the spaces is collapse, that's the default behavior

Capture12

**Comment by robintown** [comment]
Mm, well that's definitely a regression from the current behavior though, which grows the space panel up to 260px or so to fit the text in wherever possible.

Screenshot 2022-04-20 at 15-37-49 Element Element Web_Desktop

**Comment by yaya-usman** [comment]
> Mm, well that's definitely a regression from the current behavior though, which grows the space panel up to 260px or so to fit the text in wherever possible.
> 
> Screenshot 2022-04-20 at 15-37-49 Element Element Web_Desktop

Guess what, you're right. it's not a regression. when collapsed, it should maintain the length of the pane based on the longest text, let me try and resolve my patch.

**Comment by yaya-usman** [comment]
This is the supposed default behavior, sorry for the confusion : )

Capture12

**Comment by yaya-usman** [comment]
**I guess this solves the problem**

gif

File: res/css/structures/_SpacePanel.scss [hunk 1]
```diff
@@ -154,9 +159,8 @@ $activeBorderColor: $primary-content;
             align-items: center;
             border-radius: 12px;
             padding: 4px;
-            width: calc(100% - 32px);
+            width: 32px;
         }
-
```
Comment (robintown) [inline res/css/structures/_SpacePanel.scss:159]: Please remove unrelated changes in whitespace

File: res/css/structures/_SpacePanel.scss [hunk 2]
```diff
@@ -327,7 +331,6 @@ $activeBorderColor: $primary-content;
             display: block;
         }
     }
-
```
Comment (robintown) [inline res/css/structures/_SpacePanel.scss:330]: Here likewise

File: res/css/structures/_SpacePanel.scss [hunk 3]
```diff
@@ -154,9 +159,8 @@ $activeBorderColor: $primary-content;
             align-items: center;
             border-radius: 12px;
             padding: 4px;
-            width: calc(100% - 32px);
+            width: 32px;
```
Comment (robintown) [inline res/css/structures/_SpacePanel.scss:162]: This doesn't actually have the effect of setting the width to 32px, instead what it's doing is lowering the min-width of the element to allow it to collapse. So could you please replace this with `min-width: 0;` to be more clear that that is what this accomplishes?

Once you do that, you may have to set `min-width: 0;` on `.mx_SpaceTreeLevel` as well, but it should overall be the more correct approach.

File: res/css/structures/_SpacePanel.scss (hunk 2 above)
Comment (yaya-usman) [reply res/css/structures/_SpacePanel.scss:330]: the style linter is saying otherwise, it's not allowing me pass the test cases that is why

File: res/css/structures/_SpacePanel.scss (hunk 3 above)
Comment (yaya-usman) [reply res/css/structures/_SpacePanel.scss:162]: Noted

File: res/css/structures/_SpacePanel.scss (hunk 2 above)
Comment (yaya-usman) [reply res/css/structures/_SpacePanel.scss:330]: style

Comment (robintown) [reply res/css/structures/_SpacePanel.scss]: That just means that it's detecting trailing spaces on that line. If you add a newline without any spaces, it should pass the linter.

Comment (robintown) [reply res/css/structures/_SpacePanel.scss]: Ah, I see you did that

**Comment by yaya-usman** [comment]
@robintown hi  : ), my changes are still not merged yet.

**Review by robintown — APPROVED** [review]
Thanks so much for sticking with this, everything appears to be working as expected now :)
